Make the error type thread-safe

**Is your feature request related to a problem? Please describe.**

The `zfs::Error` type cannot be used with e.g. the anyhow/eyre crates that require errors to be `Send` and `Sync`:

```
error[E0277]: `*mut nvpair_sys::nvlist` cannot be sent between threads safely
   --> src/ops.rs:18:46
    |
18  |     let zfs = zfs::DelegatingZfsEngine::new()?;
    |                                              ^ `*mut nvpair_sys::nvlist` cannot be s
ent between threads safely
    |
    = help: within `libzetta::zfs::Error`, the trait `Send` is not implemented for `*mut n
vpair_sys::nvlist`
    = note: required because it appears within the type `libnv::nvpair::NvList`
    = note: required because it appears within the type `libzetta::zfs::Error`
    = note: required because of the requirements on the impl of `From<libzetta::zfs::Error
>` for `anyhow::Error`
    = note: required because of the requirements on the impl of `FromResidual<Result<Infal
lible, libzetta::zfs::Error>>` for `Result<String, anyhow::Error>`
```

because `MultiOpError(err: NvList)` contains raw pointers inside the `NvList`.

**Describe the solution you'd like**
Convert the nvlist to a Rust value… I'm looking into that

--- a/src/zfs/errors.rs
+++ b/src/zfs/errors.rs
@@ -1,7 +1,6 @@
 use crate::parsers::zfs::{Rule, ZfsParser};
-use libnv::nvpair::NvList;
 use pest::Parser;
-use std::{borrow::Cow, io, path::PathBuf};
+use std::{borrow::Cow, collections::HashMap, io, path::PathBuf};
 
 pub type Result<T, E = Error> = std::result::Result<T, E>;
 pub type ValidationResult<T = (), E = ValidationError> = std::result::Result<T, E>;
@@ -28,7 +27,7 @@ quick_error! {
         ValidationErrors(errors: Vec<ValidationError>) {
             from()
         }
-        MultiOpError(err: NvList) {
+        MultiOpError(err: HashMap<String, libnv::nvpair::Value>) {
             from()
         }
         Unimplemented {}

--- a/src/zfs/lzc.rs
+++ b/src/zfs/lzc.rs
@@ -190,7 +190,7 @@ impl ZfsEngine for ZfsLzc {
         if !errors_list_ptr.is_null() {
             let errors = unsafe { NvList::from_ptr(errors_list_ptr) };
             if !errors.is_empty() {
-                return Err(Error::from(errors));
+                return Err(Error::from(errors.into_hashmap()));
             }
         }
         match errno {
@@ -225,7 +225,7 @@ impl ZfsEngine for ZfsLzc {
         if !errors_list_ptr.is_null() {
             let errors = unsafe { NvList::from_ptr(errors_list_ptr) };
             if !errors.is_empty() {
-                return Err(Error::from(errors));
+                return Err(Error::from(errors.into_hashmap()));
             }
         }
         match errno {
@@ -265,7 +265,7 @@ impl ZfsEngine for ZfsLzc {
         if !errors_list_ptr.is_null() {
             let errors = unsafe { NvList::from_ptr(errors_list_ptr) };
             if !errors.is_empty() {
-                return Err(Error::from(errors));
+                return Err(Error::from(errors.into_hashmap()));
             }
         }
         match errno {
@@ -301,7 +301,7 @@ impl ZfsEngine for ZfsLzc {
         if !errors_list_ptr.is_null() {
             let errors = unsafe { NvList::from_ptr(errors_list_ptr) };
             if !errors.is_empty() {
-                return Err(Error::from(errors));
+                return Err(Error::from(errors.into_hashmap()));
             }
         }
         match errno {

